rpc/json/service.go

```diff
@@ -91,48 +88,30 @@ func (s *service) Subscribe(req *http.Request, args *subscribeArgs, wsConn *wsCo
 
 	// TODO(tzdybal): pass config and check subscriptions limits
 
-	q, err := tmquery.New(args.Query)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse query: %w", err)
-	}
-
 	s.logger.Debug("subscribe to query", "remote", addr, "query", args.Query)
 
 	// TODO(tzdybal): extract consts or configs
 	const SubscribeTimeout = 5 * time.Second
 	const subBufferSize = 100
-	ctx, cancel := context.WithTimeout(req.Context(), SubscribeTimeout)
-	defer cancel()
 
-	sub, err := s.client.EventBus.Subscribe(ctx, addr, q, subBufferSize)
+	eventCh, err := s.client.Subscribe(req.Context(), addr, args.Query, subBufferSize)
 	if err != nil {
 		return nil, fmt.Errorf("failed to subscribe: %w", err)
 	}
-
 	go func() {
 		for {
 			select {
-			case msg := <-sub.Out():
-				data, err := json.Marshal(msg.Data())
+			case msg := <-eventCh:
```

Comment:
It's the only case here, so the select statement can be removed to simplify the code even further - see https://github.com/celestiaorg/rollmint/pull/683.